opm/parser/eclipse/EclipseState/IOConfig/IOConfig.cpp

```diff
@@ -39,29 +40,241 @@
+
+            return dir;
+        }
+
+        inline bool is_int( const std::string& x ) {
+            auto is_digit = []( char c ) { return std::isdigit( c ); };
+
+            return !x.empty()
+                && ( x.front() == '-' || is_digit( x.front() ) )
+                && std::all_of( x.begin() + 1, x.end(), is_digit );
+        }
+    }
+
+    IOConfig::restartConfig IOConfig::rptrst( const DeckKeyword& kw, size_t step ) {
+
+        const auto& items = kw.getStringData();
+
+        /* if any of the values are pure integers we assume this is meant to be
+         * the slash-terminated list of integers way of configuring. If
+         * integers and non-integers are mixed, this is an error.
+         */
+        const auto ints = std::any_of( items.begin(), items.end(), is_int );
+        const auto strs = !std::all_of( items.begin(), items.end(), is_int );
+
+        if( ints && strs ) throw std::runtime_error(
+                "RPTRST does not support mixed mnemonics and integer list."
+            );
+
+        size_t basic = 1;
+        size_t freq = 0;
+        bool found_basic = false;
+
+        for( const auto& mnemonic : items ) {
+
+            const auto freq_pos = mnemonic.find( "FREQ=" );
+            if( freq_pos != std::string::npos ) {
+                freq = std::stoul( mnemonic.substr( freq_pos + 5 ) );
+            }
+
+            const auto basic_pos = mnemonic.find( "BASIC=" );
+            if( basic_pos != std::string::npos ) {
+                basic = std::stoul( mnemonic.substr( basic_pos + 6 ) );
+                found_basic = true;
+            }
+        }
+
+        if( found_basic ) return restartConfig( step, basic, freq );
+
+        /* If no BASIC mnemonic is found, either it is not present or we might
+         * have an old data set containing integer controls instead of mnemonics.
+         * BASIC integer switch is integer control nr 1, FREQUENCY is integer
+         * control nr 6.
+         */
+
+        /* mnemonics, but without basic and freq. Effectively ignored */
+        if( !ints ) return {};
+
+        if( items.size() >= 1 )
+            basic = std::stoul( items[ 0 ] );
+
+        // Peculiar special case in eclipse, - not documented
+        // This ignore of basic = 0 for the integer mnemonics case
+        // is done to make flow write restart file at the same intervals
+        // as eclipse for the Norne data set. There might be some rules
+        // we are missing here.
+        if( 0 == basic ) return {};
+
+        if( items.size() >= 6 ) // if frequency is set
+            freq = std::stoul( items[ 5 ] );
+
+        return restartConfig( step, basic, freq );
+    }
+
+    IOConfig::restartConfig IOConfig::rptsched( const DeckKeyword& keyword ) {
+        size_t restart = 0;
+        bool restart_found = false;
+
+        const auto& items = keyword.getStringData();
+        const auto ints = std::any_of( items.begin(), items.end(), is_int );
```

Comment:
It is, but the test itself won't get any nicer and they both have it as a precondition. It's not the worst thing in the world to duplicate, but I can look into lifting it out.